src/runner/Run_semgrep.ml

```diff
@@ -660,6 +663,14 @@ let semgrep_with_rules config ((rules, invalid_rules), rules_parse_time) =
                  is in skipped_rules *)
              target.In.rule_nums
              |> List.filter_map (fun r_num -> Hashtbl.find_opt rule_table r_num)
+             (* Don't run the extract rules
+                Note: we can't filter this out earlier because the rule ids need to be stable *)
```

Comment:
Seems quite fragile. Shouldn't the rule ids be part of the `Rule.t` type"

Reply:
And by "ids" I guess we mean "indexes" since the actual "id" of a rule is a string?

Reply:
Yeah, it's certainly why this broke in the first place 😞. The targets file interface was meant to be temporary before we ported file targeting over. I think that's still the end goal, so I'm not super invested in making it better